### DISCUSSION TITLE
Jacobian Debugger analyze_jacobian.py error

### DISCUSSION THREAD

**Discussion by Flolaffel** [idaholab/moose#21986] [Q&A Modules: General] [answered]
Hello,

I tried checking the jacobian of an input file I have. I did this before a few times and it always worked no problems. But now I get the following error:

```
Traceback (most recent call last):
  File "/home/florian/projects/intro/../moose/python/jacobiandebug/analyzejacobian.py", line 339, in <module>
    executable = findExecutable(options.executable, options.method)
  File "/home/florian/projects/intro/../moose/python/jacobiandebug/analyzejacobian.py", line 82, in findExecutable
    elif shutil.which(executable_option):
  File "/home/florian/mambaforge3/envs/moose/lib/python3.10/shutil.py", line 1444, in which
    if os.path.dirname(cmd):
  File "/home/florian/mambaforge3/envs/moose/lib/python3.10/posixpath.py", line 152, in dirname
    p = os.fspath(p)
TypeError: expected str, bytes or os.PathLike object, not NoneType
```

I reinstalled Conda recently, could this be the source of this error? How can I resolve this?

**Comment by GiudGiud** [comment, marked as answer]
Is the opt-executable built?
this looks like it s not finding an executable

**Reply by Flolaffel** [reply]
It is, I can perform simulations with intro-opt as usual

**Reply by GiudGiud** [reply]
how are you calling this? from which directory and with what command?

**Reply by Flolaffel** [reply]
Assuming I'm in my intro directory I'd call:

`../moose/python/jacobiandebug/analyzejacobian.py problems/Kontakt/segment_penalty_coulomb/2_lock_corr/jacobian/2_lock_corr_hypo_scaling.i`

Where the first part is the relative path to the script and the second part is the relative path to my input file.

**Reply by GiudGiud** [reply]
can you try specifying the executable
```
 -e EXECUTABLE, --executable=EXECUTABLE
                        The executable you would like to build an input file
                        for.  If not supplied an executable will be searched
                        for.  The searched for executable will default to the
                        optimized version of the executable (if available).
```

**Reply by Flolaffel** [reply]
That yields

`Error executing moose based application to gather DOF map`

Should I try reinstalling Conda?

**Reply by GiudGiud** [reply]
doubt that will fix.
what kind of error? can you get sort of the backtrace / where it happens?

**Reply by Flolaffel** [reply]
The line I quoted in my last reply is the only output I get when I do

`../moose/python/jacobiandebug/analyzejacobian.py problems/Kontakt/segment_penalty_coulomb/2_lock_corr/jacobian/2_lock_corr_hypo_scaling.i -e intro-opt`

**Reply by GiudGiud** [reply]
weird. I dont know of any modification.

The analyzejacobian script is just a wrapper around a bunch of options. You can run this manually and find out what s not working

**Reply by zuckarm** [reply]
i am facing this same problem. has it been resolved?
